src/cljs/ataru/hakija/application_hakukohde_handlers.cljs

```diff
@@ -36,19 +37,29 @@
   (fn [db _]
     (update-in db [:application :show-hakukohde-search] not)))
 
+
 (reg-event-db
   :application/hakukohde-query-process
   (fn [db [_ hakukohde-query]]
     (if (= hakukohde-query (get-in db [:application :hakukohde-query]))
-      (let [hakukohde-options (:options (hakukohteet-field db))
-            lang              (-> db :form :selected-language)
-            query-pattern     (re-pattern (str "(?i)" hakukohde-query))
-            results           (if (clojure.string/blank? hakukohde-query)
+      (let [lang              (-> db :form :selected-language)
+            hakukohde-options (->> (hakukohteet-field db)
+                                   :options
+                                   (sort-by #(get-in % [:label lang])))
+            query-parts       (map string/lower-case (string/split hakukohde-query #"\s+"))
+            results           (if (or (string/blank? hakukohde-query)
+                                      (< (count hakukohde-query) 2))
```

Comment:
Tän pituustarkistuksen pitäis varmaan yksinään riittää?

Reply:
Juu.

Reply:
Eiku `(count nil)` näköjään heittikin NPEn. En oo suoriltaan varma mihin arvoon tuo hakukohde-query alustettiin, mutta ehkä olisin defensiivinen ja pitäisin tarkistuksen.